applications/GeoMechanicsApplication/python_scripts/geomechanics_solver.py

```diff
@@ -137,6 +137,24 @@ def GetDefaultParameters(cls):
         this_defaults.AddMissingParameters(super().GetDefaultParameters())
         return this_defaults
 
+    def ValidateSettings(self):
+        """This function validates the settings of the solver
+        """
+
+        # Checks if scaling is used in combination with rebuild level lower than 2 and prebuild dynamics, if so it
+        # throws an error
+        if self.settings.Has("linear_solver_settings"):
+            if self.settings["linear_solver_settings"].Has("scaling") and self.settings.Has("rebuild_level"):
+                if self.settings["linear_solver_settings"]["scaling"].GetBool() and \
+                        self.settings["rebuild_level"].GetInt() < 2:
+                    raise ValueError("Scaling can only be used if rebuild level is 2")
+            if self.settings["linear_solver_settings"].Has("scaling") and self.settings.Has("prebuild_dynamics"):
+                if self.settings["linear_solver_settings"]["scaling"].GetBool() and \
+                            self.settings["prebuild_dynamics"].GetBool():
+                    raise ValueError("Scaling can not be used if prebuild dynamics is true")
```

Comment:
The check on scaling is repeated here, which is avoidable. Something like:
```
if self.settings.Has("linear_solver_settings"):
    if self.settings["linear_solver_settings"].Has("scaling"):
        if self.settings["linear_solver_settings"]["scaling"].GetBool():
            if self.settings.Has("rebuild_level"):
                if self.settings["rebuild_level"].GetInt() < 2:
                    raise ValueError("Scaling can only be used if rebuild level is 2")
            if self.settings.Has("prebuild_dynamics"):
                if self.settings["prebuild_dynamics"].GetBool():
                    raise ValueError("Scaling can not be used if prebuild dynamics is true")
```

Reply:
Preferable to have maximum 3 nested if. Can you somehow reduce it?

Reply:
thanks for the review, @WPK4FEM and @mnabideltares , these review comments are conflicting. Can you discuss with eachother which is prefered, I will then implement that.